ocs_ci/ocs/cnv/virtual_machine.py

```diff
@@ -567,7 +614,160 @@ def delete(self):
         if self.volume_interface == constants.VM_VOLUME_PVC:
             self.pvc_obj.delete()
             self.volumeimportsource_obj.delete()
-        if self.volume_interface == constants.VM_VOLUME_DV:
+        elif self.volume_interface == constants.VM_VOLUME_DV:
+            self.dv_obj.delete()
+        if self.ns_obj:
+            self.ns_obj.delete_project(project_name=self.namespace)
+
+
+class VMCloner(VirtualMachine):
+    """
+    Class for handling cloning of a Virtual Machine.
+    Inherits from VirtualMachine to have access to its attributes and methods.
+    """
+
+    def __init__(self, vm_name, namespace=None):
+        """
+        Initializes cloned vm obj
+        """
+        super().__init__(vm_name=vm_name, namespace=namespace)
+        self.source_vm_obj = None
+        self.source_pvc_name = ""
+        self.dv_cr_data_obj = self.dv_rb_data_obj = None
+
+    def clone_vm(self, source_vm_obj, volume_interface, ssh=True, verify=True):
```

Comment:
is it taken care of in case of encrypted pvc as well?

Reply:
What specifically needs to be handled here?, since encryption steps are done in the multi_cnv fixture you have created. Just pass the storageclass which has encryption configured.

Reply:
If you are creating clone PVC in same namespace as of source VM's PVC then not an issue

Reply:
clone_vm is flexible to create on same or different pvcs